storage/storage.go

```diff
@@ -139,12 +139,13 @@ func NewClient(ctx context.Context, opts ...option.ClientOption) (*Client, error
 	if err != nil {
 		return nil, fmt.Errorf("storage client: %v", err)
 	}
-	// Update readHost with the chosen endpoint.
+	// Update readHost and scheme with the chosen endpoint.
 	u, err := url.Parse(ep)
 	if err != nil {
 		return nil, fmt.Errorf("supplied endpoint %q is not valid: %v", ep, err)
 	}
 	readHost = u.Host
+	scheme = u.Scheme
```

Comment:
It looks like we set `scheme` in L114 & L123 above-- I guess this no longer does anything?

Also, is it possible for the returned endpoint from `htransport.NewClient` to have an empty scheme?

I'm thinking we should keep L114 & L123 as is and only set scheme here if `u.Scheme != ""`, just out of caution. Does that make sense?

Reply:
The endpoint for `htransport.NewClient` should always have a scheme as long as we send in the `WithDefaultEndpoint` option. But yes that makes sense, since it does not actually promise to.